src/rgw/rgw_user.cc

```diff
@@ -1333,12 +1390,10 @@ int RGWSubUserPool::execute_remove(RGWUserAdminOpState& op_state,
     return -EINVAL;
   }
 
-  if (op_state.will_purge_keys()) {
-    // error would be non-existance so don't check
-    user->keys.remove(op_state, &subprocess_msg, true);
-  }
+  // always purge all associate keys
+  user->keys.remove_subuser_keys(op_state, &subprocess_msg, defer_user_update);
```

Comment:
@yehudasa Okay, I have modified the third parameter to be defer_user_update, not always true. Please take a look at it, thank you!